client/landing/stepper/declarative-flow/internals/index.tsx

```diff
@@ -141,7 +142,7 @@ export const FlowRenderer: React.FC< { flow: Flow } > = ( { flow } ) => {
 	useSignUpStartTracking( { flow, currentStepRoute: currentStepRoute } );
 
 	return (
-		<Suspense fallback={ <StepperLoader /> }>
+		<Boot fallback={ <StepperLoader /> }>
```

Comment:
Is it possible to simply add a new component to wait for the locale to be loaded in the Suspense? This way, we won't need to reproduce the fallback behavior in the `Boot` component, leveraging its core functionality, and it probably would be simpler.

Reply:
We also wouldn't prevent the other components from loading in parallel, as opposed to your proposal, which will start loading the other components only after we have the appropriate locale.

Reply:
Do you mean do something like?
```
<Suspense fallback={}>
  <Boot>
```
If yes, the answer is no because react doesn't provide an API to trigger manually the <Suspense>.
There are weird ways to force `<Suspense>` to wait for internal components but they are not recommended https://stackoverflow.com/questions/59791769/what-is-the-react-official-position-for-throwing-promises-inside-render-function

Another issue is that capturing we are not able to capture the promise, to change it we need to implement a big refactoring.

Reply:
@gabrielcaires Thanks for the response. Regarding the time to load the component, is that something that we should worry about? My concern is that this change will add some time to start loading the other components, that's why I thinking that we could put this to load in parallel instead of blocking the other components.

Reply:
This change is not increasing the time to load a step component because we have [a code](https://github.com/Automattic/wp-calypso/blob/584cc081e1996507d686dbdfa4b04993947b1a31/client/landing/stepper/declarative-flow/internals/index.tsx#L45-L57)  that preloads the step components.

The potential increase of rendering it is that we previously rendered the step content "faster" with a possibly wrong language and then had the UI switch the content to the correct language. 

There is an initiative happening to introduce a way to track the stepper step rendering performance, so we probably can evaluate the current state and plan solutions to load the internationalization if needed.  p1715786768590259-slack-C7YPUHBB2

Reply:
I think the time to load should be pretty similar after the change, _except_ for cases where we need to switch locales. (Please sanity check me on this!)

**Case 1: No locale switch needed**

In this case, the first  render of `Boot` detects that the locales match, so we immediately trigger the transition from the `useEffect()`, which then returns the `Suspense` element and kicks off the main Stepper load sequence.

This has _some_ overhead, but it's not a lot, and should be resolved within the first render. So I think the safeguard is worth it.

**Case 2: Locale switch needed**

In this case, the first render of `Boot` detects that the locales don't match yet, so we don't load the `Suspense` yet. Once the locale's match, we'll get a re-render, and our effect will trigger the transition. However, it looks like we may only render the `Suspense` wrapper and kick off the main Stepper load sequence _after_ the transition is complete.

I did some testing with an incognito Chrome set up to mimic a slow 3G connection and `id` as the language (which is a possible user situation), and it looks like the fetches for the steps aren't really delayed by much -- the flow seems to be loaded at the same time as the language manifest file (which is what occurs before the `useLocale()`/i18n language gets switched, and the step loads are queued up before the language is fully returned.

I think this delivers a better UX, even if it may be slower.

_Side note:_ It looks like the experimental `use` function ([React docs](https://react.dev/reference/react/use)) might be helpful for us in future to have a way of deferring the display of the `Suspense` component's children until we resolve a promise relating to the locale state.

Reply:
Yes. When we have a future refactor on the switch locale we can also explore the react-router loaders.